### ISSUE TITLE
Not usable on debian jessie 

### ISSUE THREAD

**Issue by pRiVi** [neithernut/git-dit#16]
First, if $EDITOR is not defined, it won't work:

root@journaldev:/opt/git-dit# git dit new
git: 'interpret-trailers' is not a git command. See 'git --help'.
/opt/git-dit/git-dit-new: line 123: /opt/git-dit/.git/COMMIT_EDITMSG: Permission denied
The format of the provided message is faulty
root@journaldev:/opt/git-dit# 

Second, if $EDITOR is available, it hangs:

root@journaldev:/opt/git-dit# git dit new
git: 'interpret-trailers' is not a git command. See 'git --help'.
[dit][new]: f5b67c4b9ce06d5930dd929200d4e31018486ddc
root@journaldev:/opt/git-dit# git dit push origin
error: unknown option `contains'
usage: git for-each-ref [options] [<pattern>]

    -s, --shell           quote placeholders suitably for shells
    -p, --perl            quote placeholders suitably for perl
    --python              quote placeholders suitably for python
    --tcl                 quote placeholders suitably for tcl

    --count <n>           show only <n> matched refs
    --format <format>     format to use for the output
    --sort <key>          field name to sort on

error: unknown option `contains'
usage: git for-each-ref [options] [<pattern>]

    -s, --shell           quote placeholders suitably for shells
    -p, --perl            quote placeholders suitably for perl
    --python              quote placeholders suitably for python
    --tcl                 quote placeholders suitably for tcl

    --count <n>           show only <n> matched refs
    --format <format>     format to use for the output
    --sort <key>          field name to sort on

Everything up-to-date
root@journaldev:/opt/git-dit#

**Comment by neithernut** [comment]
As previously explained in the issue @pRiVi filed in his own repo (when we did not yet enable issues here), the git version shipped with Debian Jessie (2.1) is simply too old (well, Debian) and lacks some features we depend on.

This project is still just a proof of concept and will be ported away from shell-scripts anyway. Hence, we will not perform any backports at this stage. It may well be that the problem is automatically resolved during the port. If Debian provides a decent rust-setup, that is (else you may need to compile things from source).

Regarding the `EDITOR` variable, I'll file an issue here myself, as this has nothing to do with Debian per se.

**Comment by matthiasbeyer** [comment]
> It may well be that the problem is automatically resolved during the port. If Debian provides a decent rust-setup, that is (else you may need to compile things from source).

Yes, it will be resolved automatically when re-writing this in Rust, as far as I can see.

**Comment by neithernut** [comment]
`git-dit` _should_ now work on Debian Jessie. Someone should test it, though...

**Comment by neithernut** [comment]
I just compiled Rust on a Debian Jessie VM. Aside from [Rust-1.17](https://www.rust-lang.org/en-US/install.html) I had to install:
 * gcc
 * make
 * cmake
 * pkg-config
 * libssl-dev

`git-dit` appears to work. So far, I tested "fetch", "list", "show", "new" and "reply".

I'll consider `git-dit` running on Debian Jessie for now. Should other bugs surface, new issues can be opened. They are most likely rather unspecific to Debian Jessie anyways.
